app/controllers/assessments_controller.rb

```diff
@@ -417,6 +419,14 @@ def destroy
 
     @assessment.attachments.each(&:destroy)
 
+    # Delete config file copy in assessmentConfig
+    if File.exist? @assessment.config_file_path
+      File.delete @assessment.config_file_path
+    end
+    if File.exist? @assessment.config_backup_file_path
+      File.delete @assessment.config_backup_file_path
+    end
+
```

Comment:
The previous PR aimed to stop a class of user errors by first compiling the config file in order to check for any `SyntaxErrors`. However, this was not foolproof as other errors such as `NameErrors` and `ArgumentErrors` would not be caught, and when the config file is reloaded this leads to the user being locked out of the assignment / course.

The ideal solution would be to gracefully handle such errors. For now, however, the previous PR added code to create backup files so that users could at the very least revert to a last known-good configuration.

Reply:
ah that is quite smart. I will take a look at the previous pr later